src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK007ThreadHelperJTFRunAsync.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+
+namespace Microsoft.VisualStudio.SDK.Analyzers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.VisualStudio.Shell;
+
+    /// <summary>
+    /// Tasks created from <see cref="ThreadHelper.JoinableTaskFactory"/> must be awaited or joined.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSSDK007ThreadHelperJTFRunAsync : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The value to use for <see cref="DiagnosticDescriptor.Id"/> in generated diagnostics.
+        /// </summary>
+        public const string Id = "VSSDK007";
+
+        /// <summary>
+        /// A reusable descriptor for diagnostics produced by this analyzer.
+        /// </summary>
+        internal static readonly DiagnosticDescriptor Descriptor = new(
+            id: Id,
+            title: "ThreadHelper.JoinableTaskFactory.RunAsync",
+            messageFormat: "Await/join tasks created from ThreadHelper.JoinableTaskFactory.RunAsync.",
+            description: "ThreadHelper.JoinableTaskFactory.RunAsync is not safe to use for fire-and-forget tasks inside Visual Studio. Either await/join the JoinableTask, or use the JoinableTaskFactory instance from AsyncPackage or ToolkitThreadHelper.",
+            helpLinkUri: Utils.GetHelpLink(Id),
```

Comment:
We need to add a `VSSDK007.md` document to https://github.com/microsoft/VSSDK-Analyzers/tree/main/doc for this analyzer, since `GetHelpLink` will direct the users to it and it doesn't yet exist.
We should also update `index.md` to reference it. 
Can you look at the other docs and come up with something similar?

Reply:
Sure thing, I hope that's suitable. I went with a title of `Avoid ThreadHelper for fire and forget tasks` instead of the more technical `Tasks created from ThreadHelper.JoinableTaskFactory.RunAsync must be awaited or joined` since I think the majority of cases that produce this warning will be deliberate fire and forget code that happens to use ThreadHelper.

The CI build for my docs changeset has an error with dotnet.exe. Not sure what the problem is there.

`##[error]Error: The process 'C:\hostedtoolcache\windows\dotnet\dotnet.exe' failed with exit code 1`